tests/auth.test.ts

```diff
@@ -97,6 +98,110 @@ describe('account registration', () => {
     verify(emailService.sendEmailVerification(user.email, user.firstName, anyString()))
       .never();
   });
+
+  test('User cannot register with a handle that is already in use', async () => {
+    const conn = await DatabaseConnection.get();
+    const existingMember = UserFactory.fake();
+
+    await new PortalState()
+      .createUsers(existingMember)
+      .write();
+
+    const user = {
+      email: 'acm@ucsd.edu',
+      firstName: 'ACM',
+      lastName: 'UCSD',
+      password: 'password',
+      major: UserFactory.major(),
+      graduationYear: UserFactory.graduationYear(),
+      handle: existingMember.handle,
+    };
+
+    const emailService = mock(EmailService);
+    when(emailService.sendEmailVerification(user.email, user.firstName, anyString()))
+      .thenResolve();
+    const authController = ControllerFactory.auth(conn, instance(emailService));
+    const registerRequest = { user };
+    await expect(authController.register(registerRequest, FactoryUtils.randomHexString()))
+      .rejects.toThrow('This handle is already in use.');
+
+    verify(emailService.sendEmailVerification(user.email, user.firstName, anyString()))
+      .never();
+  });
+
+  test('User registration with a long name truncates handle to exactly 32 characters', async () => {
+    const conn = await DatabaseConnection.get();
+    const existingMember = UserFactory.fake();
+
+    await new PortalState()
+      .createUsers(existingMember)
+      .write();
+
+    const user = {
+      email: 'acm@ucsd.edu',
+      firstName: 'ACMACMACMACMACMACMACMACMACMACMACMACMACMACM',
+      lastName: 'UCSDUCSDUCSDUCSDUCSDUCSDUCSDUCSD',
+      password: 'password',
+      major: UserFactory.major(),
+      graduationYear: UserFactory.graduationYear(),
+    };
```

Comment:
nit: instead of hardcoding a long firstName and lastName here, let's instead generate a random string of a specified length so that it's clearer what length of the full name is. e.g.:
```ts
const user = UserRegistrationFactory.fake({
  firstName: faker.datatype.string(20),
  lastName: faker.datatype.string(20),
})
```